App/Models/Repository.cs

```diff
@@ -0,0 +1,238 @@
+        public SQLiteRepository(string connectionString)
+        {
+            connection = new SqliteConnection(connectionString);
+            connection.Open();
+            CreateUsersTable();
+            CreatePermsTable();
+            AddUser(Constants.DEFAULT_USER);
+        }
+
+        #region Permission CRUD
+        public Permissions GetTaskPermissions(User user, User other, int task_id)
+        {
+            using (var command = connection.CreateCommand())
+            {
+                command.CommandText = $"SELECT * FROM access_{user.Id} WHERE user_id = {other.Id} AND task_id = {task_id};";
+                using (SqliteDataReader reader = command.ExecuteReader())
+                {
+                    if (reader.Read())
+                    {
+                        return (Permissions)reader.GetInt32(0);
+                    }
+                    else return Permissions.None;
+                }
+            }
+        }
+        public void SetTaskPermissions(User user, User other, int task_id, Permissions perm)
+        {
+            using (var command = connection.CreateCommand())
+            {
+                if (perm == Permissions.None)
+                {
+                    command.CommandText = $"DELETE FROM access{user.Id} WHERE user_id = {other.Id} AND task_id = {task_id};";
+                    return;
+                }
+                command.CommandText = $"SELECT * FROM access_{user.Id} WHERE user_id = {other.Id} AND task_id = {task_id};";
+                using (SqliteDataReader reader = command.ExecuteReader())
+                {
+                    command.CommandText = reader.Read()
+                        ? $"UPDATE access_{user.Id} SET perm_id = {(int)perm} WHERE user_id = {other.Id} AND task_id = {task_id};"
+                        : $"INSERT INTO access_{user.Id} (user_id, task_id, perm_id) VALUES ({other.Id}, {task_id}, {(int)perm});";
+                    command.ExecuteNonQuery();
+                }
+            }
+        }
+        #endregion
+
+        #region Task CRUD
+        public void AddTask(User user, Task newTask)
+        {
+            using (var command = connection.CreateCommand())
+            {
+                command.CommandText = $"INSERT INTO tasks_{user.Id} (task_id, title, description, created, due, is_done) VALUES (" +
+                    $"{newTask.Id}, " +
+                    $"'{newTask.Title}', " +
+                    $"'{newTask.Description}', " +
+                    $"'{newTask.Created}', " +
+                    $"'{newTask.Due}', " +
+                    $"{newTask.IsDone});";
+                command.ExecuteNonQuery();
+            }
+        }
+        public void DeleteTask(User user, int task_id)
+        {
+            using (var command = connection.CreateCommand())
+            {
+                command.CommandText = $"DELETE FROM tasks_{user.Id} WHERE task_id = {task_id};";
+                command.ExecuteNonQuery();
+            }
```

Comment:
The code `using (...) {
command.CommandText = ..;
command.ExecuteNonQuery();
}` repeats very often, maybe we could abstract it somehow?

Reply:
OK, I'll try to simplify it